#! /bin/sh /usr/share/dpatch/dpatch-run
## 54_fix_sconf_mem_leak.dpatch by Daniel T Chen <crimsun@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix mem leak in pcm direct plugins (sconf must be deleted)
## DP: http://hg-mirror.alsa-project.org/alsa-lib?cs=e7949a013139;style=raw

@DPATCH@
diff -urNad alsa-lib-1.0.13~/src/pcm/pcm_dmix.c alsa-lib-1.0.13/src/pcm/pcm_dmix.c
--- alsa-lib-1.0.13~/src/pcm/pcm_dmix.c	2006-09-29 07:42:57.000000000 -0400
+++ alsa-lib-1.0.13/src/pcm/pcm_dmix.c	2007-02-04 18:54:25.000000000 -0500
@@ -1162,8 +1162,7 @@
 
 	err = snd_pcm_dmix_open(pcmp, name, &dopen, &params,
 				root, sconf, stream, mode);
-	if (err < 0)
-		snd_config_delete(sconf);
+	snd_config_delete(sconf);
 	return err;
 }
 #ifndef DOC_HIDDEN
diff -urNad alsa-lib-1.0.13~/src/pcm/pcm_dshare.c alsa-lib-1.0.13/src/pcm/pcm_dshare.c
--- alsa-lib-1.0.13~/src/pcm/pcm_dshare.c	2006-09-29 07:42:57.000000000 -0400
+++ alsa-lib-1.0.13/src/pcm/pcm_dshare.c	2007-02-04 18:54:25.000000000 -0500
@@ -898,8 +898,7 @@
 
 	err = snd_pcm_dshare_open(pcmp, name, &dopen, &params,
 				  root, sconf, stream, mode);
-	if (err < 0)
-		snd_config_delete(sconf);
+	snd_config_delete(sconf);
 	return err;
 }
 #ifndef DOC_HIDDEN
diff -urNad alsa-lib-1.0.13~/src/pcm/pcm_dsnoop.c alsa-lib-1.0.13/src/pcm/pcm_dsnoop.c
--- alsa-lib-1.0.13~/src/pcm/pcm_dsnoop.c	2006-09-29 07:42:57.000000000 -0400
+++ alsa-lib-1.0.13/src/pcm/pcm_dsnoop.c	2007-02-04 18:54:25.000000000 -0500
@@ -770,8 +770,7 @@
 
 	err = snd_pcm_dsnoop_open(pcmp, name, &dopen, &params,
 				  root, sconf, stream, mode);
-	if (err < 0)
-		snd_config_delete(sconf);
+	snd_config_delete(sconf);
 	return err;
 }
 #ifndef DOC_HIDDEN
